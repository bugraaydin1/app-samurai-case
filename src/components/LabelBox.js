const LabelBox = ({ style, selectedLanguage }) => {
	return (
		<div className="label-box">
			<input readOnly value={selectedLanguage} />
		</div>
	);
};

export default LabelBox;
