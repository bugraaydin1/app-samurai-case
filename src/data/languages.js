export const languages = [
	{
		title: "English",
		code: "en",
		enterText: "Enter text",
		translation: "Translation",
		loading: "Translating...",
		noHistory: "History not found",
		history: "Translation History",
		flag: "ğŸ‡¬ğŸ‡§",
	},
	{
		title: "TÃ¼rkÃ§e",
		code: "tr",
		enterText: "Metin giriniz",
		translation: "Ã‡eviri",
		loading: "Ã‡eviri YapÄ±lÄ±yor...",
		noHistory: "GeÃ§miÅŸ arama bulunamadÄ±",
		history: "Ã‡eviri GeÃ§miÅŸi",
		flag: "ğŸ‡¹ğŸ‡·",
	},
];
