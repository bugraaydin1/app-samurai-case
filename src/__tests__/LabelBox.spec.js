/* eslint-disable testing-library/no-render-in-setup */
import { screen, render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import LabelBox from "../components/LabelBox";
import { languages } from "../data/languages";

describe("LabelBox", () => {
	const lang = languages[0].title;

	const labelBoxComponent = <LabelBox selectedLanguage={lang} />;

	beforeEach(() => {
		render(labelBoxComponent);
	});

	afterAll(() => {
		cleanup();
	});

	it("renders correctly", () => {
		const labelBox = renderer.create(labelBoxComponent).toJSON();
		expect(labelBox).toMatchSnapshot();
	});

	it("renders with supplied language", () => {
		const input = screen.getByDisplayValue(lang);
		expect(input).toBeInTheDocument();
	});

	it("has readonly attribute", () => {
		const input = screen.getByDisplayValue(lang);
		const isReadOnly = input.hasAttribute("readonly");
		expect(isReadOnly).toBe(true);
	});
});
