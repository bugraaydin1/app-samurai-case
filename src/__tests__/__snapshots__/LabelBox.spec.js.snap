// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelBox renders correctly 1`] = `
<div
  className="label-box"
>
  <input
    readOnly={true}
    value="English"
  />
</div>
`;
